### PR TITLE
[#1896] Cleaned up EditorHelper to allow dragging/dropping actions to be defined outside of the file. Much cleaner, less duplicate code

### PR DESCRIPTION
**Pull request by mjschranz**
https://webmademovies.lighthouseapp.com/projects/65733/tickets/1896-change-the-editorhelper-drag-and-drop-code-to-pass-percentages

### CODE REVIEW

File: templates/assets/editors/editorhelper.js [hunk 1]
```diff
@@ -1,147 +1,92 @@
...
+    } //updateFunction
+
+    //Add a size chooser to the template
+    function _sizeSwitcher( e ) {
+      var selectedSize = this.options[ this.selectedIndex ].value,
+          wrapper = document.getElementById( "embed-wrapper" );
+
+      if ( EMBED_SIZES[ selectedSize ] ) {
+        wrapper.style.width = EMBED_SIZES[ selectedSize ].width + "px";
+        wrapper.style.height = EMBED_SIZES[ selectedSize ].height+ CONTROLS_HEIGHT + "px";
```
Comment (secretrobotron) [inline templates/assets/editors/editorhelper.js]: Spacing before the first `+` here.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/af206355772ac93/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/af206355772ac93/output.txt

Total script time: 0.84 mins

Server is running on http://pr710.butyr.org/templates/basic/

File: templates/assets/plugins/photo/popcorn.photo.js [hunk 2]
```diff
@@ -27,25 +27,25 @@
...
             label: "Top",
-            "default": 100
+            "default": 5
           },
           left: {
             elem: "input",
             type: "number",
             label: "Left",
-            "default": 200
+            "default": 5
```
Comment (ScottDowne) [inline templates/assets/plugins/photo/popcorn.photo.js:48]: Should probably have units: "%" like zoink does in this manifest, for all the above changes.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/7f59f2428a47e98/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/7f59f2428a47e98/output.txt

Total script time: 1.01 mins

File: templates/assets/editors/editorhelper.js [hunk 3]
```diff
@@ -1,147 +1,92 @@
...
+        wrapper.style.width = "";
+        wrapper.style.height = "";
+      }
+    }
 
-    };
-  })( window, window.jQuery );
+    butter.listen( "trackeventadded", _updateFunction );
+    butter.listen( "trackeventupdated", _updateFunction );
+
```
Comment (ScottDowne) [inline templates/assets/editors/editorhelper.js]: Remove this extra newline.

File: templates/assets/editors/editorhelper.js [hunk 4]
```diff
@@ -1,147 +1,92 @@
...
+  };
+
+
```
Comment (ScottDowne) [inline templates/assets/editors/editorhelper.js]: Remove this extra newline.

File: templates/assets/editors/editorhelper.js [hunk 5]
```diff
@@ -1,147 +1,92 @@
...
+            width: 640,
+            height: 350
+          },
+          "large": {
+            width: 853,
+            height: 470
+          },
+          "fullscreen": {
+            width: 1280,
+            height: 715
```
Comment (ScottDowne) [inline templates/assets/editors/editorhelper.js]: not a huge deal, but kinda weird how some of the json attribute names have double quotes, and some do not.

**Pull request merged**